controller/utils.go

```diff
@@ -498,7 +506,18 @@ func waitCreatingPods(
 
 	for {
 		exit := true
+
+		// go select statement chooses "randomly" when two branches are available, so it can make select ignore
+		// ctx.Done() for iterations. This check prevents to enter select if the context was already cancelled.
+		if ctx.Err() != nil {
+			logger.Warn("operation canceled/timeout waiting for rooms to be created")
+			return true, nil
+		}
+
 		select {
+		case <-ctx.Done():
```

Comment:
Since we're checking if it has been cancelled, is this still required?

Reply:
Just to prevent blocking some time waiting for next loop if canceled.